pkg/kubelet/cm/dra/manager.go

```diff
@@ -171,6 +171,9 @@ func (m *ManagerImpl) PrepareResources(pod *v1.Pod) error {
 			if reqClaim == nil {
 				return fmt.Errorf("NodePrepareResources returned result for unknown claim UID %s", claimUID)
 			}
+			if result == nil {
+				return fmt.Errorf("NodePrepareResources returned nil result for claim %s/%s", reqClaim.Namespace, reqClaim.Name)
```

Comment:
This is not an error. Just check for nil each time that `result` is used.

We need unit tests for both variants.

Reply:
ok.
what do you mean by both variants?

Reply:
I have a question. This is `NodePrepareResources` (not `NodeUnprepareResources`).

If `NodePrepareResources` successfully prepares the resources, there should be CDIDevices in result, and if it fails prepare the resources, there should be error in result.

In what case, nil result is acceptable?
I read https://github.com/kubernetes/enhancements/blob/master/keps/sig-node/3063-dynamic-resource-allocation/README.md#nodeprepareresource but couldn't find any statement about that.

Reply:
> what do you mean by both variants?

A response which maps to nil and a response which maps to an empty struct.

> In what case, nil result is acceptable?

I am unsure about a nil overall result. Perhaps play it safe and assume that nil is the same as the empty struct?

Reply:
> I still think the empty result shouldn't be allowed

I don't think we need to be strict unless there is a valid reason for that. Even if plugin doesn't return cdi devices it doesn't always mean the resource (whatever it is) is not prepared.